### PR TITLE
Implemented shield help help help help

### PR DESCRIPTION
**Pull request by thomasmmitchell**
Help dialogues for all the commands.
A bit of a framework was made in the codebase to allow for consistency of formatting and ease of change.
```
→  ./shield help list archives
shield list archives [-A] [-a] [-B] [-k] [--limit=value] [--raw] [-S] [-s] [-t]
List available backup archives

ALIASES
  list archives, ls archives

FLAGS
  -A, --after=value          Show only the archives taken after this point in time
                               Specify in the format YYYYMMDD
  -a, --all                  Show all archives, regardless of validity.
                               Equivalent to '--status=all'
  -B, --before=value         Show only the archives taken before this point in time
                               Specify in the format YYYYMMDD
  -k, --skip-ssl-validation  Disable SSL certificate validation
  --limit=value              Show only the <value> most recent archives
  --raw                      Outputs information as a JSON object
  -S, --status=value         Only show archives with the specified state of validity.
                               Accepted values are one of ['all', 'valid']
                               If not explicitly set, it defaults to 'valid'
  -s, --store=value          Show only archives sent to the specified store
  -t, --target=value         Show only archives created from the specified target

RAW OUTPUT
[
  {
    "uuid": "b4a842c5-cb61-4fa1-b0c7-08260fdc3533",
    "key": "thisisastorekey",
    "taken_at": "2016-05-18 11:02:43",
    "expires_at": "2017-05-18 11:02:43",
    "status": "valid",
    "notes": "",
    "target_uuid": "b7aa8269-008d-486a-ba1b-610ee191e4c1",
    "target_plugin": "redis-broker",
    "target_endpoint": "{\"redis_type\":\"broker\"}",
    "store_uuid": "6d52c95f-8d7f-4697-ae32-b9ce51fb4808",
    "store_plugin": "s3",
    "store_endpoint": "{\"endpoint\":\"schmendpoint\"}"
  }
]
```

### CODE REVIEW

File: cmd/shield/command.go [hunk 1]
```diff
@@ -86,25 +90,49 @@ func (c *Command) Dispatch(command string, help string, fn Handler) {
 
 	if help != "" {
 		c.help = append(c.help, []string{command, help})
+		/*		c.summary[command] = ansi.Sprintf("@G{shield %s} - %s", command, help)
+				} else {
+					c.summary[command] = ansi.Sprintf("@G{shield %s}", command)
+		*/
```
Comment (jhunt) [inline cmd/shield/command.go]: If we dont need this code, can we remove it?

File: cmd/shield/help.go [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+		Header("RAW OUTPUT")
+		ansi.Fprintf(os.Stderr, "%s\n", PrettyJSON(jsonToPrint))
+	}
+}
+
+func MessageHelp(mess string, args ...interface{}) {
+	messageToPrint = mess
+}
+
+//If MessageHelp was not called, the default summary for the command will be//printed.
```
Comment (jhunt) [inline cmd/shield/help.go]: `be//printed`?

File: tui/form.go [hunk 3]
```diff
@@ -164,5 +164,6 @@ func (f *Form) BuildContent() (string, error) {
 	if err != nil {
 		return "", fmt.Errorf("Could not marshal into JSON\nmapped input:%v\nerror:%s", c, err)
 	}
+	fmt.Println("CONTENT:\n" + string(j))
```
Comment (jhunt) [inline tui/form.go]: Why are we printing (what looks like) debugging output?

**Comment by jhunt** [comment]
Is the following formatting on purpose?

```
→  ./shield help list archives
shield list archives [-A] [-a] [-B] [-k] [--limit=value] [--raw] [-S] [-s] [-t]
List available backup archives

ALIASES
  list archives, ls archives

FLAGS
  -A, --after=value          Show only the archives taken after this point in time
                               Specify in the format YYYYMMDD
  -a, --all                  Show all archives, regardless of validity.
                               Equivalent to '--status=all'
  -B, --before=value         Show only the archives taken before this point in time
                               Specify in the format YYYYMMDD
  -k, --skip-ssl-validation  Disable SSL certificate validation
  --limit=value              Show only the <value> most recent archives
  --raw                      Outputs information as a JSON object
  -S, --status=value         Only show archives with the specified state of validity.
                               Accepted values are one of ['all', 'valid']
                               If not explicitly set, it defaults to 'valid'
  -s, --store=value          Show only archives sent to the specified store
  -t, --target=value         Show only archives created from the specified target
```

File: cmd/shield/command.go (hunk 1 above)
Comment (thomasmmitchell) [inline cmd/shield/command.go]: Yes, will do.

File: tui/form.go (hunk 3 above)
Comment (thomasmmitchell) [inline tui/form.go]: It was debugging output - I have two different personal notes somewhere to remember to remove it and I forgot... whoops. Good catch!

File: cmd/shield/help.go (hunk 2 above)
Comment (thomasmmitchell) [inline cmd/shield/help.go]: Something tells me I tried to type a word in ex mode. Will fix.

**Comment by thomasmmitchell** [comment]
Yes, its on purpose - but to be honest I'm not 100% happy with it. I tried a few different styles to keep it from being a giant block of description text where its difficult to tell where one ends and the next begins. Open to alternative opinions.

**Comment by jhunt** [comment]
Left-align and insert an empty line if the previous flag stanza was a multi-line block?  Avoids the striping of unconditional line break addition, but breaks up the larger blocks quite nicely.

File: tui/form.go (hunk 3 above)
Comment (jhunt) [inline tui/form.go]: I guess that was your third note?  ^_^

**Comment by thomasmmitchell** [comment]
```
→  ./shield help edit store
shield edit store <store> [-k] [--raw]
Modify an existing backup store. The UUID of the store will remain the same after modification.

ALIASES
  edit store, update store

FLAGS
  <store>                    A string partially matching the name of a single store
                             or a UUID exactly matching the UUID of a store.
                             Not setting this value explicitly will default it to the empty string.

  -k, --skip-ssl-validation  Disable SSL certificate validation
  --raw                      Takes input as a JSON object from standard input
                             Outputs the resultant store info as a JSON object.
                             Suppresses interactive dialogues and confirmation.

RAW INPUT
{
  "endpoint": "{\"endpoint\":\"schmendpoint\"}",
  "name": "AnotherStore",
  "plugin": "s3",
  "summary": "A Test Store"
}

RAW OUTPUT
{
  "uuid": "355ccd3f-1d2f-49d5-937b-f4a12033a0cf",
  "name": "AnotherStore",
  "summary": "A Test Store",
  "plugin": "s3",
  "endpoint": "{\"endpoint\":\"schmendpoint\"}"
}
```

**Pull request merged**